src/test/java/bio/overture/ego/controller/PolicyControllerTest.java

```diff
@@ -0,0 +1,295 @@
+    val groupId = entityGenerator.setupGroup("GroupPolicyAdd").getId().toString();
+
+    val entity = new HttpEntity<String>("WRITE", headers);
+    val response =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/permission/group/%s", policyId, groupId)),
+            HttpMethod.POST,
+            entity,
+            String.class);
+
+    val responseStatus = response.getStatusCode();
+    assertThat(responseStatus).isEqualTo(HttpStatus.OK);
+    // TODO: Fix it so that POST returns JSON, not just random string message
+
+    val getResponse =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/groups", policyId)),
+            HttpMethod.GET,
+            new HttpEntity<String>(null, headers),
+            String.class);
+
+    val getResponseStatus = getResponse.getStatusCode();
+    val getResponseJson = MAPPER.readTree(getResponse.getBody());
+    val groupPermissionJson = getResponseJson.get(0);
+
+    assertThat(getResponseStatus).isEqualTo(HttpStatus.OK);
+    assertThat(groupPermissionJson.get("id").asText()).isEqualTo(groupId);
+    assertThat(groupPermissionJson.get("mask").asText()).isEqualTo("WRITE");
+  }
+
+  @Test
+  @SneakyThrows
+  public void deleteGroupPermission() {
+    val policyId = entityGenerator.setupSinglePolicy("DeleteGroupPermission").getId().toString();
+    val groupId = entityGenerator.setupGroup("GroupPolicyDelete").getId().toString();
+
+    val entity = new HttpEntity<String>("WRITE", headers);
+    val response =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/permission/group/%s", policyId, groupId)),
+            HttpMethod.POST,
+            entity,
+            String.class);
+
+    val responseStatus = response.getStatusCode();
+    assertThat(responseStatus).isEqualTo(HttpStatus.OK);
+    // TODO: Fix it so that POST returns JSON, not just random string message
+
+    val deleteResponse =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/permission/group/%s", policyId, groupId)),
+            HttpMethod.DELETE,
+            new HttpEntity<String>(null, headers),
+            String.class);
+
+    val deleteResponseStatus = deleteResponse.getStatusCode();
+    assertThat(deleteResponseStatus).isEqualTo(HttpStatus.OK);
+
+    val getResponse =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/groups", policyId)),
+            HttpMethod.GET,
+            new HttpEntity<String>(null, headers),
+            String.class);
+
+    val getResponseStatus = getResponse.getStatusCode();
+    val getResponseJson = (ArrayNode) MAPPER.readTree(getResponse.getBody());
+
+    assertThat(getResponseStatus).isEqualTo(HttpStatus.OK);
+    assertThat(getResponseJson.size()).isEqualTo(0);
+  }
+
+  @Test
+  @SneakyThrows
+  public void addUserPermission() {
```

Comment:
rename test to `associatePermissionsWithUser_ExistingEntitiesButNoRelationship_Success`